repository-module/src/main/java/com/gmail/zagurskaya/online/cash/repository/impl/RoleRepositoryImpl.java

```diff
@@ -0,0 +1,57 @@
+package com.gmail.zagurskaya.online.cash.repository.impl;
+
+import com.gmail.zagurskaya.online.cash.repository.RoleRepository;
+import com.gmail.zagurskaya.online.cash.repository.exception.RoleRepositoryException;
+import com.gmail.zagurskaya.online.cash.repository.model.Role;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Repository;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository
+public class RoleRepositoryImpl implements RoleRepository {
+
+    private static Logger logger = LoggerFactory.getLogger(RoleRepositoryImpl.class);
+
+    @Override
+    public List<Role> getRoles(Connection connection) {
+        String sql = String.format(
+                "SELECT * FROM `roles`");
+        List<Role> roleList = new ArrayList<>();
+        try (PreparedStatement prepared = connection.prepareStatement(sql)) {
+            ResultSet resultSet = prepared.executeQuery(sql);
+            return getRolesFromResult(resultSet);
+        } catch (SQLException e) {
+            logger.error(e.getMessage(), e);
+            throw new RoleRepositoryException("Database exception during  getRoles ", e);
+        }
+    }
+
+    @Override
+    public Role getRole(Connection connection, Long id) {
+        String sql = "SELECT * FROM `roles`  WHERE `id` = ? ";
+        List<Role> roleList = new ArrayList<>();
+        try (PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            return getRolesFromResult(resultSet).get(0);
```

Comment:
Need to use separate method got getting ONE result. You can get Indexofboundexception.